piet-direct2d/src/lib.rs

```diff
@@ -14,6 +14,8 @@ mod text;
 use std::borrow::Cow;
 use std::ops::Deref;
 
+use associative_cache::{AssociativeCache, Capacity1024, HashDirectMapped, RoundRobinReplacement};
```

Comment:
Oh, and something like HashFourWay is going to be a better choice here, otherwise if two colors hash to the same value it's going to bounce.